### PR TITLE
MAINT: Upgrade to spin 0.7

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

**Comment by charris** [comment]
A lot has been removed from `.spin/cmds.py`. Why? And are there replacements?

**Comment by stefanv** [comment]
A lot of the functionality developed in NumPy has been ported to spin itself.

File: .spin/cmds.py [hunk 1]
```diff
@@ -465,29 +419,6 @@ def ipython(ctx, ipython_args):
...
-    If you'd like to expand shell variables, like `$PYTHONPATH` in the example
-    above, you need to provide a single, quoted command to `run`:
-
-    spin run 'echo $SHELL && echo $PWD'
-
-    On Windows, all shell commands are run via Bash.
-    Install Git for Windows if you don't have Bash already.
-    """
-    ctx.invoke(build)
-    ctx.forward(meson.run)
```
Comment (mattip) [inline .spin/cmds.py:490]: The [upstream docstring](https://github.com/scientific-python/spin/blob/b5ea9af686c20db254ae3f278d3defc626add2c1/spin/cmds/util.py#L9) does not mention the `PYTHONPATH`. Is that intentional?

File: .spin/cmds.py [hunk 2]
```diff
@@ -193,51 +193,6 @@ def test(ctx, pytest_args, markexpr, n_jobs, tests, verbose):
...
-        code_prefix = 'import sys; sys.path.pop(0); '
-
-    if code:
-        PYTHON_ARGS = ['-c', code_prefix + code]
-        gdb_args += ['--args', sys.executable] + PYTHON_FLAGS + PYTHON_ARGS
-
-    gdb_cmd = ['gdb', '-ex', 'set detach-on-fork on'] + gdb_args
-    util.run(gdb_cmd, replace=True)
-
-
```
Comment (mattip) [inline .spin/cmds.py:240]: This seems identical to [upstream](https://github.com/scientific-python/spin/blob/b5ea9af686c20db254ae3f278d3defc626add2c1/spin/cmds/meson.py#L319)

File: .spin/cmds.py (hunk 1 above)
Comment (seberg) [reply .spin/cmds.py:490]: A bit confusing, but the `click` command upstream equivalent is here and mentions it: https://github.com/scientific-python/spin/blob/b5ea9af686c20db254ae3f278d3defc626add2c1/spin/cmds/meson.py#L462

**Comment by seberg** [comment]
I'll be optimistic and merge it thanks Stefan.  I guess we will all have to update spin :).

**Pull request merged**